src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
Our `ChangesetCooker` is and stored on `SyncHistory`. `SyncHistory` is created at the same time as the `shared_group` and is owned by the `SharedRealm`. I believe there is no lifetime issues with this reference to the cooker.

There is also a `ChangesetCooker` used by `Sync::Client` which it gets via a `SyncHistory` object via its `m_file_access` member. But I don't see where this would get initialized with our cooker instance so I don't think this is an issue (although i am not yet certain).